File: ExamenRepaso/ExamenRepaso/PrediccionDia.vb


Namespace ExamenRepaso

    Public Class PrediccionDia

        Private Sub NotifyPropertyChanged(<CallerMemberName()> Optional ByVal propertyName As String = Nothing)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
        End Sub


        Private _dia As String
        Public Property Dia() As String
            Get
                Return _dia
            End Get
            Set(ByVal value As String)
                If (value! = _dia) Then{
            _dia = value
                NotifyPropertyChanged("Dia")
            }

        End Set
        End Property

        Private _prediccion As String
        Public Property Prediccion() As String
            Get
                Return _prediccion
            End Get
            Set(ByVal value As String)
                If (value! = _prediccion) Then{
            _prediccion = value
                NotifyPropertyChanged("Prediccion")
            }
        End Set
        End Property

        Private _minimaFarenheit As Integer
        Public Property MinimaFarenheit() As Integer
            Get
                Return _minimaFarenheit
            End Get
            Set(ByVal value As Integer)
                If (value! = _minimaFarenheit) Then{
            _minimaFarenheit = value
                NotifyPropertyChanged("MinimaFarenheit")
            }
        End Set
        End Property

        Private _maximaFarenheit As Integer
        Public Property MaximaFarenheit() As Integer
            Get
                Return _maximaFarenheit
            End Get
            Set(ByVal value As Integer)
                If (value! = _maximaFarenheit) Then{
            _maximaFarenheit = value
                NotifyPropertyChanged("MaximaFarenheit")
            }
        End Set
        End Property

        Public Sub New(dia As String, prediccion As String, minima As Integer, maxima As Integer)
            Me.Dia = dia
            Me.Prediccion = prediccion
            Me.MinimaFarenheit = minima
            Me.MaximaFarenheit = maxima
        End Sub

        Public Shared Function ObtenerDatos() As ObservableCollection(Of PrediccionDia)
            Dim resultado As New ObservableCollection(Of PrediccionDia)

            resultado.Add(New PrediccionDia("Lunes", "sol", 44, 60))
            resultado.Add(New PrediccionDia("Martes", "nubes", 55, 70))
            resultado.Add(New PrediccionDia("Miércoles", "sol", 35, 55))
            resultado.Add(New PrediccionDia("Jueves", "nubes", 48, 65))
            resultado.Add(New PrediccionDia("Viernes", "lluvia", 37, 50))
            resultado.Add(New PrediccionDia("Sábado", "nubes", 30, 50))
            resultado.Add(New PrediccionDia("Domingo", "nieve", 28, 40))

            Return resultado
        End Function

    End Class

File: ExamenRepaso/PrediccionDia.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExamenRepaso
{
    class PrediccionDia : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler PropertyChanged;
        public void NotifyPropertyChanged(string propertyName)
        {
            this.PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }
        private string _dia;

        public string Dia
        {
            get { return _dia; }
            set {
                if(value != _dia)
                {
                    _dia = value;
                    NotifyPropertyChanged("Dia");
                }
                
            }
        }
        private string _prediccion;

        public string Prediccion
        {
            get { return _prediccion; }
            set
            {
                if (value != _prediccion)
                {
                    _prediccion = value;
                    NotifyPropertyChanged("Prediccion");
                }

            }
        }

        private int _minimaFarenheit;
        public int MinimaFarenheit
        {
            get { return _minimaFarenheit; }
            set
            {
                if (value != _minimaFarenheit)
                {
                    _minimaFarenheit = value;
                    NotifyPropertyChanged("MinimaFarenheit");
                }

            }
        }

        private int _maximaFarenheit;
        public int MaximaFarenheit
        {
            get { return _maximaFarenheit; }
            set
            {
                if (value != _maximaFarenheit)
                {
                    _maximaFarenheit = value;
                    NotifyPropertyChanged("MaximaFarenheit");
                }

            }
        }

        public PrediccionDia(string dia,string prediccion, int minima, int maxima)
        {
            this.Dia = dia;
            this.Prediccion = prediccion;
            this.MinimaFarenheit = minima;
            this.MaximaFarenheit = maxima;
        }

        public static ObservableCollection<PrediccionDia> ObtenerDatos()
        {
            ObservableCollection<PrediccionDia> dias = new ObservableCollection<PrediccionDia>();
            dias.Add(new PrediccionDia("Lunes", "sol", 44, 60));
            dias.Add(new PrediccionDia("Martes", "nubes", 55, 70));
            dias.Add(new PrediccionDia("Miércoles", "sol", 35, 55));
            dias.Add(new PrediccionDia("Jueves", "nubes", 48, 65));
            dias.Add(new PrediccionDia("Viernes", "lluvia", 37, 50));
            dias.Add(new PrediccionDia("Sábado", "nubes", 30, 50));
            dias.Add(new PrediccionDia("Domingo", "nieve", 28, 40));
            return dias;
        }

    }
}
